R/fb_league_stats.R

```diff
@@ -0,0 +1,150 @@
+#' @importFrom purrr map_dfr
+#' @importFrom dplyr mutate
+#' @importFrom readr type_convert
+.fb_single_league_stats <- function(url, team_or_player) {
+
+  session <- worldfootballr_chromote_session(url)
+  tables <- worldfootballr_html_table(session)
+
+  idx <- if(team_or_player == "player") { 3 } else { 1:2 }
+
+  purrr::map_dfr(
+    idx,
+    ~{
+      table <- tables[[.x]]
+
+      renamed_table <- .rename_fb_cols(table)
+
+      renamed_table <- if (team_or_player == "player") {
+        renamed_table[renamed_table$Rk != "Rk", ]
+      } else {
+        renamed_table %>%
+          dplyr::mutate(
+            "Team_or_Opponent" = ifelse(.x == 1, "team", "opponent"),
+            .before = 1
+          )
+
+      }
+
+      suppressMessages(
+        readr::type_convert(
+          renamed_table,
+          guess_integer = TRUE,
+          na = "",
+          trim_ws = TRUE
+        )
```

Comment:
i could have maybe re-used the `mutate_at` + `cols_to_transform` stuff in `.clean_advanced_stat_table`, but the players df has non-numeric columns for Nationality and Position which were being incorrectly converted to numerics. `readr::type_convert` does a good job of correctly guessing column types imo (all columns are initially characters due to the raw first row having partial column names initially)